File: output/llama3.1-8b_IO_1/avatar/Java/Python/codeforces_67_B.py
import sys

class Codeforces67B:
    def __init__(self):
        self.sc = input().split()
        self.n = int(self.sc[0])
        self.k = int(self.sc[1])

    def main(self):
        bb = list(map(int, self.sc[2:]))
        aa = [0] * self.n
        m = 0
        for a in range(self.n-1, -1, -1):
            j = 0
            while j < len(bb) and bb[j] > 0:
                if aa[j] >= a + self.k:
                    bb[j] -= 1
                j += 1
            for j_ in range(m, j, -1):
                aa[j_] = aa[j_-1]
            aa[j] = a
            m = max(m, j)
        print(*[str(x+1) for x in aa], sep=' ')

if __name__ == "__main__":
    o = Codeforces67B()
    o.main()

File: output/llama3.1-8b_IO_1/avatar/Python/Java/codeforces_67_B.java
import java.util.Scanner;

public class codeforces_67_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        int n = Integer.parseInt(input.split(" ")[0]);
        int k = Integer.parseInt(input.split(" ")[1]);

        int[] M = new int[n];
        for (int i = 0; i < n; i++) {
            M[i] = Integer.parseInt(scanner.nextLine());
        }

        int m = 0;
        for (int j = 0; j < n; j++) {
            int p = -1;
            for (int l = 0; l < n; l++) {
                if (M[l] == 0) {
                    p = l;
                    break;
                }
            }
            if (p != -1) {
                System.out.print((p + 1) + " ");
                for (int l = Math.max(0, p - k); l <= p; l++) {
                    M[l]--;
                }
                M[p]--;
            }
        }
    }
}
